src/utils/types.ts

```diff
@@ -1 +1,19 @@
+import { z } from "zod";
+import { Types } from "mongoose";
+
+export const commentSchema = z.object({
+    author: z.instanceof(Types.ObjectId),
+    post: z.instanceof(Types.ObjectId),
+    date: z.date().default(new Date()),
+    content: z.string(),
+    likes: z.number().default(0),
+    replyTo: z.instanceof(Types.ObjectId).nullable().default(null),
```

Comment:
Passing around ObjectId's is annoying with server actions, so let's just take in strings as inputs instead. Also need optional fields to handle comment inputs.
```suggestion
    author: z.string().transform(id => new Types.ObjectId(id)),
    post: z.string().transform(id => new Types.ObjectId(id)),
    date: z.date().optional().default(new Date()),
    content: z.string(),
    likes: z.number().optional().default(0),
    replyTo: z.string().transform(id => new Types.ObjectId(id)).optional(),
```